react/modules/gaEvents.ts

```diff
@@ -260,6 +261,23 @@ export function beginCheckout(eventData: BeginCheckoutData) {
   updateEcommerce(eventName, { ecommerce: data })
 }
 
+export function addShippingInfo(eventData: AddShippingInfoData) {
+  const eventName = 'add_shipping_info'
+
+  const { currency, value, items: eventDataItems, shippingTier } = eventData
+
+  const { items } = formatCartItemsAndValue(eventDataItems ?? [])
+
+  const data = {
+    currency,
+    value,
```

Comment:
Do you think we can calculate the `value` using the same logic as the `add_payment_info` event? Both actions usually happen during checkout, so they feel similar.

I mentioned in another comment that `value` will probably not come ready inside the input. We'll calculate it from the items.

I simplified the `eventDataItems ?? []` because it shouldn't be optional (also mentioned in another comment in the types file).

```suggestion
  const { currency, items: eventDataItems, shippingTier } = eventData

  const { items, totalValue } = formatCartItemsAndValue(eventDataItems ?? [])

  const data = {
    currency,
    value: totalValue,
```

Reply:
Thinking about value, in this case (add_shipping_info) does makes more sense for it correspond to the shipping value and not to the value of products?